pumpkin/src/command/commands/cmd_clear.rs

```diff
@@ -0,0 +1,129 @@
+use std::sync::Arc;
+
+use async_trait::async_trait;
+use pumpkin_core::text::color::NamedColor;
+use pumpkin_core::text::TextComponent;
+
+use crate::command::args::arg_entities::EntitiesArgumentConsumer;
+use crate::command::args::{Arg, ConsumedArgs};
+use crate::command::tree::CommandTree;
+use crate::command::tree_builder::{argument, require};
+use crate::command::{CommandExecutor, CommandSender, InvalidTreeError};
+use crate::entity::player::Player;
+use InvalidTreeError::InvalidConsumptionError;
+
+const NAMES: [&str; 1] = ["clear"];
+const DESCRIPTION: &str = "Clear yours or targets inventory.";
+
+const ARG_TARGET: &str = "target";
+
+async fn clear_player(target: &Arc<Player>) -> usize {
+    let mut inventory = target.inventory.lock().await;
+
+    // There may be a better way to do this with something like all_slots() or .clear on Player.inventory
+    let mut items_count: usize = 0;
+    for slot in 0..=45 {
+        if let Some(item) = inventory.get_slot(slot).unwrap() {
+            items_count += item.item_count as usize;
+            inventory.set_slot(slot as u16, None, true).unwrap();
+            target
+                .send_inventory_slot_update(&mut inventory, slot)
+                .await
+                .unwrap();
+        }
+    }
+    items_count
+}
+
+struct ClearExecutor;
+
+#[async_trait]
+impl CommandExecutor for ClearExecutor {
+    async fn execute<'a>(
+        &self,
+        sender: &mut CommandSender<'a>,
+        _server: &crate::server::Server,
+        args: &ConsumedArgs<'a>,
+    ) -> Result<(), InvalidTreeError> {
+        let Some(Arg::Entities(targets)) = args.get(&ARG_TARGET) else {
+            return Err(InvalidConsumptionError(Some(ARG_TARGET.into())));
+        };
+
+        let target_count = targets.len();
+
+        let mut items_count = 0;
+        for target in targets {
+            let item_count = clear_player(target).await;
+            items_count += item_count;
+        }
+
+        let msg = if target_count == 1 {
+            let target = targets.first().unwrap();
+            if items_count == 0 {
+                TextComponent::text_string(format!(
+                    "No items were found on player {}",
+                    target.gameprofile.name
+                ))
+                .color_named(NamedColor::Red)
+            } else {
+                TextComponent::text_string(format!(
+                    "Removed {} item(s) on player {}",
+                    items_count, target.gameprofile.name
+                ))
+            }
+        } else if items_count == 0 {
```

Comment:
Don't make it an else, It could be also one target with none items

Reply:
One target with no item is handled above this else if

Reply:
The message conditional statements are a bit weird, I was forced to do it like this because of a clippy warning.